Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputMapper.cpp

```diff
@@ -317,6 +319,19 @@ namespace AzToolsFramework
         return false;
     }
 
+    AZ::s32 QtEventToAzInputMapper::GetPriority() const
+    {
+        return AzFramework::InputChannelEventListener::GetPriorityLast();
+    }
+
+    void QtEventToAzInputMapper::OnInputChannelEvent(const AzFramework::InputChannel& inputChannel, bool& hasBeenConsumed)
+    {
+        if (m_enabled && hasBeenConsumed)
+        {
+            m_lastConsumedInputChannelEvent = &inputChannel;
+        }
+    }
```

Comment:
It seems dangerous to store a pointer to a `const&` reference parameter.

What if a temporary AzFramework::InputChannel is created when invoke the this event?

Reply:
This is a virtual function, we can't change the signature. It should be fine because all InputChannels are created at input device creation which happens at startup, although it isn't outside the realm of possibility that InputChannels could have a different lifecycle under some circumstances, so your concern (while unlikely to transpire) is valid. Storing a temporary InputChannel isn't really feasible because they can only be constructed using a reference to their parent InputDevice, but we could easily store the corresponding AzFramework::InputChannelId and achieve the same goal.

Reply:
I don't believe that's possible on the temporary, @bosnichd would be able to confirm.  As for changing the argument type would be a large sweeping API change.